packages/conventional-changelog-cli/cli.js

```diff
@@ -112,6 +112,7 @@ try {
 
   if (flags.config) {
     config = require(resolve(process.cwd(), flags.config));
+    options.config = config;
```

Comment:
Could you just assign the required value to `options.config` and then use `options.config` elsewhere? That way there is only a single source of truth for the loaded configuration.

Reply:
I'm not quite sure I follow. The main reason that I'm setting `options.config` here is because if config is a promise then `writerOpts`, `parserOpts`, and `gitRawCommitsOpts` won't be populated and can't be sent downstream. The other option is to resolve the promise in `conventional-changelog-cli` from the config, and then handle assignment of all of the opts after the resolution and then call `conventionalChangelog`.

The config gets resolved as a promise here: https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-core/lib/merge-config.js#L93

I modeled this change off of: https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog/index.js#L9

That is where presets are loaded to be resolved as a promise. And I don't see any other place that sets `options.config` so the only conflict that would appear would be if a custom config AND a preset was set. In this case, perhaps extending the custom config over top of the preset is something that should be done.

There are several ways to slice it and I'll leave it to you guys to decide as I'm not as familiar with the code base.

Reply:
Thank you for the explanation. I didn't realize you were relying on the promise resolution behavior of `conventional-changelog-core` to resolve `options.config`. This line makes a lot more sense now.